src/mock.js

```diff
@@ -82,6 +82,17 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     return customMock;
   }
 
+  function assignResolveType(type) {
+    const fieldType = getNullableType(type);
+    const namedFieldType = getNamedType(fieldType);
+
+    if (namedFieldType instanceof GraphQLUnionType ||
+        namedFieldType instanceof GraphQLInterfaceType
+    ) {
+      namedFieldType.resolveType = (data, context, info) => info.schema.getType(data.typename);
```

Comment:
what I can do is to change it to:

```js
namedFieldType.resolveType = (data, context, info) => namedFieldType.resolveType(data, context, info) || info.schema.getType(data.typename);
```

so that it uses the custom one only if the default one returns falsey. Not sure if this is an hack :D

plus checking `preserveResolvers` before, of course